optiboot/bootloaders/optiboot/optiboot.c

```diff
@@ -589,22 +607,23 @@ int main(void) {
         // This is the reset vector page. We need to live-patch
         // the code so the bootloader runs first.
         //
-        // Move RESET vector to WDT vector
+        // Move RESET vector to 'save' vector
 	// Save jmp targets (for "Verify")
 	rstVect0_sav = buff[rstVect0];
 	rstVect1_sav = buff[rstVect1];
-	wdtVect0_sav = buff[wdtVect0];
-	wdtVect1_sav = buff[wdtVect1];
+	saveVect0_sav = buff[saveVect0];
+	saveVect1_sav = buff[saveVect1];
 
 	// Instruction is a relative jump (rjmp), so recalculate.
-	uint16_t vect=rstVect0_sav+(rstVect1_sav<<8);
-        vect -= WDT_vect_num;
-        // Move RESET jmp target to WDT vector
-        buff[wdtVect0] = vect & 0xff;
-        buff[wdtVect1] = vect >> 8;
+	uint16_t vect=(rstVect0_sav & 0xff) | ((rstVect1_sav & 0x0f)<<8); //calculate 12b displacement
+	vect = (vect-save_vect_num) & 0x0fff; //substract 'save' interrupt position and wrap around 4096
+        // Move RESET jmp target to 'save' vector
+        buff[saveVect0] = vect & 0xff;
+        buff[saveVect1] = (vect >> 8) | 0xc0; //
         // Add rjump to bootloader at RESET vector
-        buff[0] = (((uint16_t)main) & 0xFFF) & 0xFF; // rjmp 0x1d00 instruction
-	buff[1] =  ((((uint16_t)main) & 0xFFF) >> 8) | 0xC0;
+        vect = ((uint16_t)main) &0x0fff; //WARNIG: this works as long as 'main' is in first section
```

Comment:
Using "main" here is going to conflict with another patch in the pipeline that uses the .init2 section.
(Yeah, that was a problem with the old code too.  Just reminding myself.)